File: week_02/day_1/bank_account.rb
class BankAccount

  attr_reader(:holder_name, :amount, :type)
  attr_writer(:holder_name, :amount, :type)

  def initialize(holder_name, amount, type)
    @holder_name = holder_name
    @amount = amount
    @type = type
  end

#The methods below are replaced by the attr_writer above
  # def set_holder_name(name)
  #   @holder_name = name
  # end
  #
  # def set_account_amount(amount)
  #   @amount = amount
  # end
  #
  # def set_account_type(type)
  #   @type = type
  # end


#The methods below are replaced by the attr_reader above
  # def holder_name
  #   return @holder_name
  # end
  #
  # def amount
  #   return @amount
  # end
  #
  # def type
  #   return @type
  # end

end

File: projects/csharp/day_1/class_with_testing/class_with_testing/BankAccount.cs
using System;
namespace class_with_testing
{
    public class BankAccount
    {
        //string name;
        //int amount;
        //string type; 

        public string Name
        {
            get;
            set;
        }

        public int Amount
        {
            get;
            set;
        }

        public string Type
        {
            get;
            set;
        }

        public BankAccount(string name, int amount, string type)
        {
            this.Name = name;
            this.Amount = amount;
            this.Type = type;
        }

    }
}
